### PR TITLE
Add grind option to select Java version

### PR DESCRIPTION
**Pull request by jzhuge**
Also include the commit to remove Google Maven Central mirror, otherwise grind will not work with Hadoop trunk. Unfortunately this might affect performance and reliability. A better solution is to deploy a repository manager inside dist-test cluster.

### CODE REVIEW

File: grind/bin/grind [hunk 1]
```diff
@@ -355,6 +356,11 @@ class TestRunner:
                             type=int,
                             default=default_timeout,
                             help="Per-task timeout. The default timeout is %s seconds." % default_timeout)
+        parser.add_argument('--java-version',
+                            type=int,
+                            help="Select Java version. Valid choices include 7 and 8. " +
```
Comment (umbrant) [inline grind/bin/grind]: We can validate with argparse's choices, e.g.

parser.add_argument("-v", "--verbosity", type=int, choices=[0, 1, 2],
                    help="increase output verbosity")

Can also set a default value.

File: grind/bin/grind [hunk 2]
```diff
@@ -355,6 +356,11 @@ class TestRunner:
...
+                                 "If not specified, project config option \"java_version\" will apply. " +
```
Comment (umbrant) [inline grind/bin/grind]: we also need to update docs/grind.md with this new config information.

File: grind/python/disttest/isolate.py [hunk 3]
```diff
@@ -60,11 +60,12 @@ class Isolate:
 
     __ISOLATE_NAME = """disttest.isolate"""
 
-    def __init__(self, project_root, output_dir,
+    def __init__(self, project_root, output_dir, java_version=None,
                  include_modules=None, exclude_modules=None, include_patterns=None, exclude_patterns=None,
                  cache_dir=None, extra_deps=None, maven_flags=None, maven_repo=None, verbose=False):
         logger.info("Using output directory " + output_dir)
         self.output_dir = output_dir
+        self.java_version = java_version
```
Comment (umbrant) [inline grind/python/disttest/isolate.py]: I'd like some kind of assert that this is not null for safety, since otherwise we won't generate a valid run script.

File: docs/grind.md [hunk 4]
```diff
@@ -133,6 +137,7 @@ Like the `grind config` command, `pconfig` will generate a default config to the
 As an example, here is Hadoop's `.grind_project.cfg`. Hadoop tests expect a few extra directories to be created, and we also need to upload the native libraries that are not picked up by the Maven Dependency Plugin.
 
         [grind]
+        java_verson = 8
```
Comment (umbrant) [inline docs/grind.md]: typo here, would be good to regenerate using "grind pconfig"

File: grind/bin/grind [hunk 5]
```diff
@@ -184,7 +186,9 @@ class ProjectConfig(BaseConfig):
 
     _section = "grind"
     _required = {}
+    global default_java_version
```
Comment (umbrant) [inline grind/bin/grind]: don't need to say global unless you're modifying this

File: grind/bin/grind [hunk 6]
```diff
@@ -355,6 +359,12 @@ class TestRunner:
                             type=int,
                             default=default_timeout,
                             help="Per-task timeout. The default timeout is %s seconds." % default_timeout)
+        global supported_java_versions
```
Comment (umbrant) [inline grind/bin/grind]: same for these globals

File: grind/python/disttest/isolate.py [hunk 7]
```diff
@@ -97,14 +98,20 @@ def _generate_run_script_contents(self):
         exit $rc
     fi
 }
-export JAVA7_BUILD=true
-. /opt/toolchain/toolchain.sh
+
 run ./unpack.sh # Generated by packager
 source environment.source # Init runtime environment
+"""
+        logging.info("Select Java version {java_version}"
```
Comment (umbrant) [inline grind/python/disttest/isolate.py]: Could we move the log print out of where we construct the run script?

File: grind/python/disttest/isolate.py [hunk 8]
```diff
@@ -97,14 +98,20 @@ def _generate_run_script_contents(self):
...
+                     .format(java_version=self.java_version))
+        contents += """
+jdks=(/usr/lib/jvm/java-{java_version}-*)
+export JAVA_HOME="${{jdks[0]}}"
+""".format(java_version=self.java_version)
+
+        contents += """
 run which mvn
 run mvn -version
-run which java
```
Comment (umbrant) [inline grind/python/disttest/isolate.py]: why are these removed?

File: grind/python/disttest/isolate.py (hunk 3 above)
Comment (umbrant) [inline grind/python/disttest/isolate.py]: since we have argparse, safe to ignore this now I think.

File: grind/python/disttest/isolate.py (hunk 8 above)
Comment (jzhuge) [inline grind/python/disttest/isolate.py]: "mvn -version" prints Java version and home already:

> $ mvn -version
> Apache Maven 3.3.9 (bb52d8502b132ec0a5a3f4c09453c07478323dc5; 2015-11-10T08:41:47-08:00)
> Maven home: /Users/jzhuge/apache-maven-3.3.9
> Java version: 1.7.0_79, vendor: Oracle Corporation
> Java home: /Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home/jre
> Default locale: en_US, platform encoding: UTF-8
> OS name: "mac os x", version: "10.12.2", arch: "x86_64", family: "mac"
>

**Comment by umbrant** [comment]
Nice work here, thanks for the contribution John!
